File: pinyougou-manager-web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//分页 
	this.findPage=function(searchEntity,page,rows){
		return $http.post('../goods/findPage.do?page='+page+'&rows='+rows, searchEntity);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../goods/delete.do?ids='+ids);
	}
	
	//删除
	this.deleGoods=function(ids){
		return $http.get('../goods/deleGoods.do?ids='+ids);
	}
	
	// 审核与驳回
	this.updateAuditStatus = function(ids,status) {
		return $http.get('../goods/updateAuditStatus.do?status='+status+"&ids="+ids);
	}
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbGoods;
import com.pinyougou.pojo.TbItem;
import com.pinyougou.povo.Goods;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Goods goods);
	
	
	/**
	 * 修改
	 */
	public void update(Goods goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum,int pageSize);


	/** 审核与驳回
	 * @param ids
	 * @param status
	 */
	public void updateAuditStatus(Long[] ids, String status);


	/** 删除商品
	 * @param ids
	 */
	public void deleGoods(Long[] ids);


	/** 上下架操作
	 * @param goods
	 */
	public void updateMaeketable(TbGoods goods);
	
	/**根据商品ID和状态查询Item表信息  
	 * @param ids
	 * @param status
	 * @return
	 */
	public List<TbItem> findItemListByGoodsIdandStatus(Long[] ids,String status);
	
}
